### PR TITLE
Add tests for kyber swap

### PR DESCRIPTION
**Pull request by manuelwedler**
This PR adds integration tests for token swaps on Kyber. It adds a node package for running a Ganache blockchain. This was the easiest way to test the Kyber integration because Kyber provides a snapshot of a Ganache chain with all necessary contracts deployed. So I didn't have to do any deployment by myself. (Using the official Kyber deployment on Ropsten would have needed a faucet that could be used)

### CODE REVIEW

File: tests/fake_blockchain/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,5320 @@
...
+        "fd-slicer": "~1.1.0"
+      }
+    },
+    "zeppelin-solidity": {
+      "version": "1.12.0",
+      "resolved": "https://registry.npmjs.org/zeppelin-solidity/-/zeppelin-solidity-1.12.0.tgz",
+      "integrity": "sha512-dgjPPnTmx14hAbTeOpTKemDeDCDdwglS0nquOAJG8h5o9zlb43FZafQSrMlIUUSp1EisDZfehrp5loGEYXHZBA=="
+    }
+  }
+}
```
Comment (konradkonrad) [inline tests/fake_blockchain/package-lock.json:5320]: holy cow....that's insane...

File: tests/integration/test_kyber.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    body = dict(jsonrpc="2.0", method="evm_revert", params=[snapshot_id], id=1)
+    requests.post(f"http://localhost:{PORT}", json=body)
+
+
+@pytest.fixture
+def test_account(monkeypatch):
+    monkeypatch.setattr(Account, "generate_private_key", lambda: to_bytes(WALLET_PRIVATE_KEY))
+    account = Account.create(TESTING_KEYSTORE_FOLDER, "test_password")
+    yield account
+    account.keystore_file_path.unlink()
```
Comment (konradkonrad) [inline tests/integration/test_kyber.py:69]: I'd suggest a contant/fixture for the password, i.e.

```suggestion
@pytest.fixture
def test_password():
    return "test_password"

@pytest.fixture
def test_account(monkeypatch, test_password):
    monkeypatch.setattr(Account, "generate_private_key", lambda: to_bytes(WALLET_PRIVATE_KEY))
    account = Account.create(TESTING_KEYSTORE_FOLDER, test_password)
    yield account
    account.keystore_file_path.unlink()
```
```suggestion
    account = Account.create(TESTING_KEYSTORE_FOLDER, "test_password")
```

File: tests/integration/test_kyber.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+
+def test_cannot_buy_without_eth(test_account, patch_kyber_support, kyber):
+    tx = {
+        "to": "0x0000000000000000000000000000000000000000",
+        "from": test_account.address,
+        "value": test_account.get_ethereum_balance(kyber.w3).as_wei,
+        "gasPrice": 0,
+    }
+    kyber.w3.eth.sendTransaction(tx)
```
Comment (konradkonrad) [inline tests/integration/test_kyber.py:132]: I would add
```suggestion
    kyber.w3.eth.sendTransaction(tx)
    assert test_account.get_ethereum_balance(kyber.w3).as_wei == 0, "account not empty"
```
to make sure the account is actually drained before proceeding with the test.

**Review by konradkonrad — APPROVED** [review]
looks good to me!
I have a few suggestions. and please share your experience with `ganache` once you feel familiar with it!

File: tests/integration/test_kyber.py (hunk 2 above)
Comment (manuelwedler) [reply tests/integration/test_kyber.py:69]: Nice!

File: tests/integration/test_kyber.py (hunk 3 above)
Comment (manuelwedler) [reply tests/integration/test_kyber.py:132]: Wouldn't it make more sense to wait for the transaction to get included in a block?

**Pull request merged**